tests/unittests/tests-mtd/tests-mtd.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2016 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+#include <string.h>
+#include <errno.h>
+
+#include "embUnit.h"
+
+#include "mtd.h"
+#include "board.h"
+
+#ifdef MTD_0
+#define _dev MTD_0
+#else
+#ifndef SECTOR_COUNT
+#define SECTOR_COUNT 8
+#endif
+#ifndef PAGE_PER_SECTOR
+#define PAGE_PER_SECTOR 4
+#endif
+#ifndef PAGE_SIZE
+#define PAGE_SIZE 255
+#endif
+
+static uint8_t dummy_sector[PAGE_PER_SECTOR * PAGE_SIZE];
+
+static int init(mtd_dev_t *dev)
+{
+    memset(dummy_sector, 0xff, sizeof(dummy_sector));
+    return 0;
+}
+
+static int read(mtd_dev_t *dev, void *buff, uint32_t addr, uint32_t size)
+{
+    (void)dev;
+    (void)addr;
+    memcpy(buff, dummy_sector, size);
+
+    return size;
+}
+
+static int write(mtd_dev_t *dev, const void *buff, uint32_t addr, uint32_t size)
+{
+    (void)dev;
+    (void)addr;
+    memcpy(dummy_sector, buff, size);
+
+    return size;
+}
+
+static int erase(mtd_dev_t *dev, uint32_t addr, uint32_t size)
+{
+    (void)dev;
+    (void)addr;
+
+    if (size % (PAGE_PER_SECTOR * PAGE_SIZE) != 0) {
+        return -EOVERFLOW;
+    }
+
+    return 0;
+}
+
+static int power(mtd_dev_t *dev, enum mtd_power_state power)
+{
+    (void)dev;
+    (void)power;
+    return 0;
+}
+
+const mtd_desc_t driver = {
+    .init = init,
+    .read = read,
+    .write = write,
+    .erase = erase,
+    .power = power,
+};
+
+static mtd_dev_t _dev = {
+    .driver = &driver,
+    .sector_count = SECTOR_COUNT,
+    .pages_per_sector = PAGE_PER_SECTOR,
+    .page_size = PAGE_SIZE,
+};
+#endif
+
+static void setup_teardown(void)
+{
+    mtd_dev_t *dev = (mtd_dev_t*) &_dev;
+
+    mtd_erase(dev, 0, dev->pages_per_sector * dev->page_size);
+}
+
+static void test_mtd_init(void)
+{
+    mtd_dev_t *dev = (mtd_dev_t*) &_dev;
+
+    int ret = mtd_init(dev);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+}
+
+static void test_mtd_erase(void)
+{
+    mtd_dev_t *dev = (mtd_dev_t*) &_dev;
+
+    int ret = mtd_erase(dev, 0, dev->pages_per_sector * dev->page_size);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    ret = mtd_erase(dev, 0, dev->pages_per_sector);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+}
+
+static void test_mtd_write_read(void)
```

Comment:
Some additional tests for write would be useful:
 - offsets other than 0
 - write out of bounds (testing both starting offset too big and offset+size out of bounds)
 - For memories which behave like flash: test the AND nature of the write, e.g. write 0xee, then 0x33, and the result should be 0x22